actions/store_playlist_info_MOD.js

```diff
@@ -0,0 +1,99 @@
+module.exports = {
+
+  name: 'Store Playlist Info',
+  section: 'YouTube Tools',
+
+  subtitle (data) {
+    return 'Store YouTube playlist information.'
+  },
+
+  variableStorage (data, varType) {
+    const type = parseInt(data.storage)
+    if (type !== varType) return
+    const dataType = 'Playlist Info'
+    return ([data.varName, dataType])
+  },
+
+  fields: ['query', 'info', 'storage', 'varName'],
+
+  html (isEvent, data) {
+    return `
+<div style="width: 90%;">
+  Playlist URL:<br>
+  <input id="query" class="round" type="text">
+</div><br>
+<div style="padding-top: 8px; width: 60%;">
+  Options:
+  <select id="info" class="round">
+    <option value="0" selected>Video Data List</option>
+    <option value="1">Video URL List</option>
+    <option value="2">Video Name List</option>
+    <option value="3">Video Duration List</option>
+    <option value="4">Video ID List</option>
+  </select>
+</div><br>
+<div style="padding-top: 8px;">
+  <div style="float: left; width: 35%;">
+    Store In:<br>
+    <select id="storage" class="round">
+      ${data.variables[1]}
+    </select>
+  </div>
+  <div id="varNameContainer" style="float: right; width: 60%;">
+    Variable Name:<br>
+    <input id="varName" class="round" type="text">
+  </div>
+</div>`
+  },
+
+  init () {},
+
+  async action (cache) {
+    const data = cache.actions[cache.index]
+    const storage = parseInt(data.storage)
+    const varName = this.evalMessage(data.varName, cache)
+    const query = this.evalMessage(data.query, cache)
+    const INFO = parseInt(data.info)
+    const ytlist = require('youtube-playlist')
+    const url = query
+    let result = 5
+
+    const urls = await ytlist(url, 'url')
+    const urllist = JSON.stringify(urls)
+
+    const names = await ytlist(url, 'name')
+    const namelist = JSON.stringify(names)
+
+    const datas = await ytlist(url, ['id', 'name', 'url'])
+    const datalist = JSON.stringify(datas)
+
+    const durationis = await ytlist(url, 'duration')
+    const durationlist = JSON.stringify(durationis)
+
+    const ids = await ytlist(url, 'id')
+    const idlist = JSON.stringify(ids)
```

Comment:
You only need to fetch all of these when their options are selected, so put them inside each case on the switch and adapt the code accordingly